File: android-frontend/SmartMedicine/view/appView/src/main/java/com/czy/appview/view/order/OrderViewPagerEnum.java
package com.czy.appview.view.order;

public enum OrderViewPagerEnum {
    // 预约订单
    APPOINTMENT_ORDER(0),

    // 购物订单
    PURCHASE_ORDER(1)
    ;

    public final int value;
    OrderViewPagerEnum(int value) {
        this.value = value;
    }

    // code -> o
    public static OrderViewPagerEnum getByValue(int value) {
        for (OrderViewPagerEnum orderViewPagerEnum : values())
        {
            if (orderViewPagerEnum.value == value)
            {
                return orderViewPagerEnum;
            }
        }
        return APPOINTMENT_ORDER;
    }
}

File: android-frontend/SmartMedicine/view/appView/src/main/java/com/czy/appview/view/order/OrderViewPagerEnum.kt
package com.czy.appview.view.order

enum class OrderViewPagerEnum(val index: Int) {

    // 预约订单
    APPOINTMENT_ORDER(0),

    // 购物订单
    PURCHASE_ORDER(1);

    // 根据 index 获取枚举值
    companion object {
        fun getEnumByIndex(index: Int): OrderViewPagerEnum {
            return values().find { it.index == index } ?: APPOINTMENT_ORDER
        }

        // 获取枚举总数
        fun getCount(): Int {
            return values().size
        }
    }
}